### PR TITLE
Add timezone to containers

### LINKED ISSUE #1154: Timezone in containers
Hello,

I have seen that timezones are not set for containers. This causes logs to have wrong timestamps and email "received" headers have wrong timezones.

It should be possible to set the timezone for all containers. The setting should be exposed and described in the .env file.

### PR DESCRIPTION
**Pull request by DjVinnii**
## What type of PR?

Enhancement

## What does this PR do?
This PR adds the tzdata package so that the environment variable `TZ` can be used to set the timezone of containers.

### Related issue(s)
- closes #1154 

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by Diman0** [comment]
Did you forget adapting the Dockerfiles of rainloop and roundcube?
https://github.com/Mailu/Mailu/blob/master/webmails/rainloop/Dockerfile
https://github.com/Mailu/Mailu/blob/master/webmails/roundcube/Dockerfile

For the documentation, please mention that the value for TZ can be found on https://en.wikipedia.org/wiki/List_of_tz_database_time_zones. The value  "TZ database name" can be used from the table on that page.

**Comment by DjVinnii** [comment]
@Diman0 My bad. I added the tzdata to the webmails as well.

**Comment by ghostwheel42** [comment]
LGTM

I'm wondering if creating /etc/localtime and /etc/timezone would be useful?
```
cp /usr/share/zoneinfo/${TZ} /etc/localtime
echo "${TZ}" >  /etc/timezone
```
I think tzdata could be uninstalled after this (smaller image).

**Comment by DjVinnii** [comment]
@ghostwheel42 Tzdata has to be present to make use of /etc/localtime and /etc/timezone.

**Review by Diman0 — DISMISSED** [review]
Since we introduce a new ENV variable, it should be added to the config manager with a default value. I'm referring to this file:
https://github.com/Mailu/Mailu/blob/5e212ea46d869dfc97fc22a71ee705b5c6a46901/core/admin/mailu/configuration.py#L9

You could add the line
``'TZ': 'Etc/UTC',``

It would help users a lot, if we added the new setting to the mailu.env file (with a comment above it). This is another hint for users that they can change the timezone. To alter the mailu.env file for compose/stack you can alter:
https://github.com/Mailu/Mailu/blob/master/setup/flavors/compose/mailu.env
It makes sense to me to add it in the section "# Common configuration variables".
E.g.
```
# Timezone for the Mailu containers. See this link for all possible values https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
TZ=Etc/UTC
```

**Comment by ghostwheel42** [comment]
@DjVinnii 
> Tzdata has to be present to make use of /etc/localtime and /etc/timezone

That's not exactly correct. It has to be present when copying, but after this the whole set of timezones is not necessary anymore. See https://wiki.alpinelinux.org/wiki/Setting_the_timezone

This is: install - copy/echo - remove

The nginx Dockerfile would look like this:

```
ENV TZ=Etc/UTC

# Image specific layers under this line
RUN set -eo \
 && apk add --no-cache certbot nginx nginx-mod-mail openssl curl tzdata \
 && pip3 install watchdog \
 && cp /usr/share/zoneinfo/${TZ} /etc/localtime \
 && echo "${TZ}" >/etc/timezone \
 && apk del tzdata

```

**Comment by DjVinnii** [comment]
@ghostwheel42 That's weird. I tried this, however the timezone reverted back to UTC after removing tzdata. I will take a look at it this afternoon.

**Comment by ghostwheel42** [comment]
Yes. that's weird. I've tested it using only the nginx container and it worked as expected.
Maybe the `ENV TZ=Etc/UTC` makes the difference.

**Comment by DjVinnii** [comment]
It just came to mind that your way of setting the timezone wouldn't allow users to set the timezone without building the image themselves.

**Comment by ghostwheel42** [comment]
> It just came to mind that your way of setting the timezone wouldn't allow users to set the timezone without building the image themselves.

Yes, you are correct. I didn't see this - my bad. tzdata needs to remain inside the image.
I still do not know if setting `/etc/localtime` or `/etc/timezone` should be done too.
Currently I don't see any need, but some (obscure) software added later could rely on this.

**Comment by ghostwheel42** [comment]
If it's necessary to change the /etc files it can be done in a later PR in the entrypoint.

File: webmails/roundcube/Dockerfile [hunk 1]
```diff
@@ -21,12 +21,14 @@ ENV ROUNDCUBE_URL https://github.com/roundcube/roundcubemail/releases/download/1
 
 ENV CARDDAV_URL https://github.com/mstilkerich/rcmcarddav/releases/download/v4.1.2/carddav-v4.1.2.tar.gz
 
+ENV TZ Etc/UTC
```
Comment (ghostwheel42) [inline webmails/roundcube/Dockerfile:24]: Please add this line to all Dockerfiles to set a default. 

The 'echo TZ' needs to be moved to entrypoint (jinja template php.ini)

**Review by ghostwheel42 — DISMISSED** [review]

File: webmails/roundcube/Dockerfile (hunk 1 above)
Comment (DjVinnii) [reply webmails/roundcube/Dockerfile:24]: When setting the timezone in roundcube.ini (jinja), does the timezone still has to be set in timezone.ini?

**Comment by ghostwheel42** [comment]
@DjVinnii Thanks for changing this. 
Now there are some discussions in the matrix dev channel about using any other timezone except UTC making Mail debugging really painful.
It's true: a real MTA should always use UTC to make the logs consistent. 
And the best solution for the different timestamps in the logs would be to change the hosts timezone to UTC.

I think this PR can still be merged, but maybe you could add another comment to the env file and/or docs like so: "warning: if you are observing different timestamps in your log files you should change your hosts timezone to UTC instead of changing TZ to your local timezone. Using UTC allows easy log correlation with remote MTAs."

**Comment by ghostwheel42** [comment]
bors try

**Comment by nextgens** [comment]
Please move it to the "Advanced settings" section too; it's definitely not "Common configuration" and not something we should encourage people to change.

Daylight Saving Time makes all of this very difficult to correlate: DST isn't the same everywhere (https://en.wikipedia.org/wiki/Daylight_saving_time_by_country). The solution is trivial: stick to GMT/Zulu time everywhere.

**Comment by ghostwheel42** [comment]
bors try

**Review by ghostwheel42 — APPROVED** [review]
local test successful. LGTM

**Review by Diman0 — APPROVED** [review]
LGTM. Thank you for your contribution.

**Pull request merged**